library/src/main/java/bio/terra/stairway/Control.java

```diff
@@ -0,0 +1,583 @@
+            .submitted(rs.getTimestamp("submit_time").toInstant())
+            .completed(rs.getTimestamp("completed_time"))
+            .exception(rs.getString("serialized_exception"))
+            .stairwayId(rs.getString("stairway_id"));
+        flightList.add(flight);
+      }
+    }
+    return flightList;
+  }
+
+  private List<Control.KeyValue> keyValueQuery(NamedParameterPreparedStatement statement)
+      throws SQLException {
+
+    List<Control.KeyValue> keyValueList = new ArrayList<>();
+    try (ResultSet rs = statement.getPreparedStatement().executeQuery()) {
+      while (rs.next()) {
+        KeyValue keyValue = new KeyValue();
+        keyValue.key(rs.getString("key")).value(rs.getString("value"));
+        keyValueList.add(keyValue);
+      }
+    }
+    return keyValueList;
+  }
+
+  // -- Stairway methods --
+
+  public List<String> listStairways() throws Exception {
+    return stairwayInstanceDao.getList();
+  }
+
+  // -- POJOs for returning results --
+
+  // Flight - represents a row from the "flight" table
+  public static class Flight {
+    private String flightId;
+    private String className;
+    private FlightStatus status;
+    private Instant submitted;
+    private Instant completed;
+    private String exception; // Un-deserialized exception text
+    private String stairwayId;
+
+    public String getFlightId() {
+      return flightId;
+    }
+
+    public Flight flightId(String flightId) {
+      this.flightId = flightId;
+      return this;
+    }
+
+    public String getClassName() {
+      return className;
+    }
+
+    public Flight className(String className) {
+      this.className = className;
+      return this;
+    }
+
+    public FlightStatus getStatus() {
+      return status;
+    }
+
+    public Flight status(FlightStatus status) {
+      this.status = status;
+      return this;
+    }
+
+    public Instant getSubmitted() {
+      return submitted;
+    }
+
+    public Flight submitted(Instant submitted) {
+      this.submitted = submitted;
+      return this;
+    }
+
+    public Optional<Instant> getCompleted() {
+      return Optional.ofNullable(completed);
+    }
+
+    public Flight completed(Timestamp completed) {
+      this.completed = Optional.ofNullable(completed).map(Timestamp::toInstant).orElse(null);
+      return this;
+    }
+
+    public Optional<String> getException() {
+      return Optional.ofNullable(exception);
+    }
+
+    public Flight exception(String exception) {
+      this.exception = exception;
+      return this;
+    }
+
+    public Optional<String> getStairwayId() {
+      return Optional.ofNullable(stairwayId);
+    }
+
+    public Flight stairwayId(String stairwayId) {
+      this.stairwayId = stairwayId;
+      return this;
+    }
+  }
+
+  // Holds flight map data as a pair of strings; either inputs or working map
+  // Comparable by key for an alphabetical display
+  public static class KeyValue implements Comparable<KeyValue> {
```

Comment:
optional nit: consider `FlightMapEntry`? Little more specific about what this is for.

Reply:
I like it. Thanks!